packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv.ts

```diff
@@ -0,0 +1,257 @@
+import { Server } from "http";
+import { Server as SecureServer } from "https";
+
+import type { Express } from "express";
+import { Optional } from "typescript-optional";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+import {
+  ICactusPlugin,
+  ICactusPluginOptions,
+  IPluginWebService,
+  IWebServiceEndpoint,
+  PluginAspect,
+} from "@hyperledger/cactus-core-api";
+
+import { KeyVaultSecret, SecretClient } from "@azure/keyvault-secrets";
+import {
+  UsernamePasswordCredential,
+  DefaultAzureCredential,
+} from "@azure/identity";
+
+import { PluginKeychainAzureKvMock } from "../../test/typescript/mock/plugin-keychain-azure-kv-mock";
+
+// TODO: Writing the getExpressRequestHandler() method for
+// GetKeychainEntryEndpointV1 and SetKeychainEntryEndpointV1
+// import { GetKeychainEntryEndpointV1 } from "./web-services/get-keychain-entry-endpoint-v1";
+// import { SetKeychainEntryEndpointV1 } from "./web-services/set-keychain-entry-endpoint-v1";
+
+export enum AzureCredentialType {
+  LocalFile = "LOCAL_FILE",
+  InMemory = "IN_MEMORY",
+}
+
+export interface IAzureInMemoryCredentials {
+  azureTenantId: string;
+  azureClientId: string;
+  azureUsername: string;
+  azurePassword: string;
+}
+
+export interface IPluginKeychainAzureKvOptions extends ICactusPluginOptions {
+  logLevel?: LogLevelDesc;
+  keychainId: string;
+  instanceId: string;
+  azureEndpoint: string;
+  azureCredentialType: AzureCredentialType;
+  azureInMemoryCredentials?: IAzureInMemoryCredentials;
+}
+
+export class PluginKeychainAzureKv implements ICactusPlugin, IPluginWebService {
```

Comment:
@jagpreetsinghsasan Also implement the keychain interface from the core api package (as long as I remember correctly that we have one)

Reply:
Is this something new that we have added, because I dont see this one implemented for vault plugin as well.

Reply:
Any update on this @petermetz ?

Reply:
@jagpreetsinghsasan Oops, missed this comment earlier (helps if you tag me, then GitHub highlights it specifically)

Double checked it, it's been around since last year so not too new: 
packages/cactus-core-api/src/main/typescript/plugin/keychain/i-plugin-keychain.ts